iree/turbine/kernel/wave/templates/gqa_vanilla_attention.py

```diff
@@ -81,13 +81,21 @@ def low_precision_clamp(source_reg, upper_bound):
     constraints += [tkw.WorkgroupConstraint(B, BLOCK_B, 2)]
     constraints += [tkw.WorkgroupConstraint(H, BLOCK_H, 3)]
     constraints += [tkw.WorkgroupConstraint(H_KV, BLOCK_H, 3, primary=False)]
-    constraints += [
-        tkw.TilingConstraint(
-            N_KV,
-            BLOCK_N_KV,
-            iters=sympy.Min((WORKGROUP_0 + 1) * BLOCK_N_Q, N_KV) // BLOCK_N_KV,
-        )
-    ]
+    if is_causal:
+        constraints += [
+            tkw.TilingConstraint(
+                N_KV,
+                BLOCK_N_KV,
+                iters=sympy.Min((WORKGROUP_0 + 1) * BLOCK_N_Q, N_KV) // BLOCK_N_KV,
```

Comment:
Nice!
If not already added, shall we include a comment about why `iters` are omitted/included based on `is_causal` flag?